### PR TITLE
generate manifest file from the collected_sources

### PR DESCRIPTION
**Pull request by yash-zededa**
The PR will do the below:-

- Creation of the manifest file using shell script
- Archiving the manifest file in the collected_source.tar.gz
- Content of manifest file.

```
alpine,acl.2.3.1-r0,1ea87358d76d9e006cfc081d6a79dc636ee0ed26,alpine/acl.2.3.1-r0.1ea87358d76d9e006cfc081d6a79dc636ee0ed26
alpine,alpine-baselayout.3.2.0-r16,8a8c96a0ea2fcd824c361aa4438763fa33ee8ca0,alpine/alpine-baselayout.3.2.0-r16.8a8c96a0ea2fcd824c361aa4438763fa33ee8ca0
alpine,alpine-baselayout.3.2.0-r23,348653a9ba0701e8e968b3344e72313a9ef334e4,alpine/alpine-baselayout.3.2.0-r23.348653a9ba0701e8e968b3344e72313a9ef334e4
alpine,apk-tools.2.12.7-r0,03c6b1b0ec20f417cee0f3c1d56795b27871b1c5,alpine/apk-tools.2.12.7-r0.03c6b1b0ec20f417cee0f3c1d56795b27871b1c5
alpine,apk-tools.2.12.9-r3,34d90ac8388e88126893f5d27ea35d304e65e5ab,alpine/apk-tools.2.12.9-r3.34d90ac8388e88126893f5d27ea35d304e65e5ab
alpine,attr.2.5.1-r1,c0b3bcef6de7d3ae5b326994c3282d8b4f524451,alpine/attr.2.5.1-r1.c0b3bcef6de7d3ae5b326994c3282d8b4f524451
```

### CODE REVIEW

**Comment by deitch** [comment]
A few requests before a formal review:

* can you squash the commits into a single commit?
* can you include here (in the comments) a few sample lines of what the output csv will look like?
* there are many yetus complains

File: Makefile [hunk 1]
```diff
@@ -634,7 +635,8 @@ $(COLLECTED_SOURCES): $(ROOTFS_TAR) $(GOSOURCES)| $(INSTALLER)
 		do								 		   \
 			$(GOSOURCES) sources -d $$(dirname $$i) --recursive --out $(TMP_COLLECTED_SOURCES)/go; \
 		done;	
-	tar cvf - $(TMP_COLLECTED_SOURCES) | gzip -9 - > $(COLLECTED_SOURCES)
+	tar -cf $(COLLECTED_SOURCES) -C $(TMP_COLLECTED_SOURCES) . 
+	bash tools/generate-sources-manifests.sh $(COLLECTED_SOURCES) $(COLLECTED_SOURCES_MANIFESTFILE)
```
Comment (deitch) [inline Makefile:639]: Can we move this up a line? By which I mean, _before_ creating the tar file with all of the sources, can we generate the manifest, with all of the paths relative to the root (i.e. `$(TMP_COLLECTED_SOURCES)`), and then have the manifest _included_ in the tar file? So right before you tar, you end up with:

```
$(TMP_COLLECTED_SOURCES)/go          # go sources
$(TMP_COLLECTED_SOURCES)/alpine    # alpine sources
$(TMP_COLLECTED_SOURCES)/kernel    # kernel sources
$(TMP_COLLECTED_SOURCES)/manifest.csv
```

and then the tar file includes all of those directories _and_ the `manifest.csv`.

The result of which will be a single tar file, with the manifest inside.

**Review by deitch — CHANGES_REQUESTED** [review]
Besides the yetus, request to include the manifest inside the tar file.

File: Makefile (hunk 1 above)
Comment (yash-zededa) [reply Makefile:639]: I will make the required changes.

**Comment by yash-zededa** [comment]
> A few requests before a formal review:
> 
> * can you squash the commits into a single commit?
> * can you include here (in the comments) a few sample lines of what the output csv will look like?
> * there are many yetus complains

@deitch 

based on reviews we might have changes in  the script or in the makefile. I prefer to squash at once when all the changes are reviewed and approved.

will include few lines of output from the manifest file.

will check and fix the yetus stuff if possible.

**Comment by shjala** [comment]
@yash-zededa for yetus, copy-pasting the script [here](https://www.shellcheck.net/) might help.

File: tools/generate-sources-manifests.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        commit=$(echo $file| rev | cut -d "." -f1 | rev)
+        name_version=$(echo $file | rev | cut -d "." -f2- | rev )
+        echo "alpine",$name_version,$commit,"alpine/$file" >> $MANIFESTS_FILE
+done
+}
+
+## go packages
+
+go_manifests() {
+cd $BASEDIR/go
```
Comment (eriknordmark) [inline tools/generate-sources-manifests.sh:23]: This cd approach implicitly assumes that $1 hence BASEDIR is an absolute path, which I'm not sure the Makefile caller guarantees.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
See comments plus complaints from shellcheck and blanks from the yetus run.

File: tools/generate-sources-manifests.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
```
Comment (eriknordmark) [inline tools/generate-sources-manifests.sh:1]: Missing Copyright and SPDX-license. See other scripts.

Comment (yash-zededa) [reply tools/generate-sources-manifests.sh:1]: Added the copyright and the licence info.

File: tools/generate-sources-manifests.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        name_version=$(echo $file | rev | cut -d "." -f2- | rev )
+        echo "alpine",$name_version,$commit,"alpine/$file" >> $MANIFESTS_FILE
+done
+}
+
+## go packages
+
+go_manifests() {
+cd $BASEDIR/go
+for file in `ls $BASEDIR/go`
```
[the remark this answers is not in the document]
Comment (yash-zededa) [reply tools/generate-sources-manifests.sh]: Latest changes were not pushed to the remote. Yes, I have removed the BASEDIR as it was obsolete

File: tools/generate-sources-manifests.sh (hunk 2 above)
Comment (yash-zededa) [reply tools/generate-sources-manifests.sh]: This is fixed.

File: Makefile (hunk 1 above)
Comment (yash-zededa) [reply Makefile]: changes done.

**Comment by deitch** [comment]
I think you need to rebase.

**Comment by yash-zededa** [comment]
> I think you need to rebase.

Yes, I have synced the fork

**Comment by deitch** [comment]
Yetus is clean! You do need to sign the DCO though

**Review by deitch — APPROVED** [review]

File: tools/generate-sources-manifests.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+# This script is used to create a manifest file for alpine,go and kernel packages.
+# Collect EVE's kernel sources from the upstream.
+# Example usage
+#  generate-sources-manifests.sh <collected_sources_path> <manifest_file_path>
```
Comment (eriknordmark) [inline tools/generate-sources-manifests.sh:8]: That seems like the old usage since the Makefile only passes one argument

**Review by eriknordmark — APPROVED** [review]
LGTM; nit about the PR description and the script usage line.

**Comment by yash-zededa** [comment]
> Your sample manifest in the description is very confusing as it is a single line. Is that what the script outputs? If not, can you fix the description?

```
alpine,acl.2.3.1-r0,1ea87358d76d9e006cfc081d6a79dc636ee0ed26,alpine/acl.2.3.1-r0.1ea87358d76d9e006cfc081d6a79dc636ee0ed26
```

**Pull request merged**